src/Illuminate/Support/Collection.php

```diff
@@ -296,6 +296,34 @@ public function containsStrict($key, $value = null)
         return in_array($key, $this->items, true);
     }
 
+    /**
+     * Join all items from the collection using a string. The final items can use a separate glue string.
+     *
+     * @param  string  $glue
+     * @param  string  $finalGlue
+     * @return string
+     */
+    public function join($glue, $finalGlue = '')
+    {
+        if ($finalGlue === '') {
+            return $this->implode($glue);
+        }
+
+        if ($this->count() === 0) {
+            return '';
+        }
+
+        if ($this->count() === 1) {
```

Comment:
Can you cache the result of `count` method?

Reply:
Makes sense, done.